tests/unit/iam_auth_tests.py

```diff
@@ -0,0 +1,262 @@
+            port_specified=False,
+            domain='localhost',
+            domain_specified=False,
+            domain_initial_dot=False,
+            path="/",
+            path_specified=True,
+            secure=True,
+            expires=int(time.time() + expires_secs),
+            discard=False,
+            comment=None,
+            comment_url=None,
+            rest={'HttpOnly': None},
+            rfc2109=True)
+
+    @mock.patch('cloudant._common_util.ClientSession.request')
+    def test_iam_get_access_token(self, m_req):
+        m_response = mock.MagicMock()
+        m_response.json.return_value = MOCK_OIDC_TOKEN_RESPONSE
+        m_req.return_value = m_response
+
+        iam = IAMSession(MOCK_API_KEY, 'http://127.0.0.1:5984')
+        access_token = iam._get_access_token()
+
+        m_req.assert_called_once_with(
+            'POST',
+            iam._token_url,
+            auth=('bx', 'bx'),
+            headers={'Accepts': 'application/json'},
+            data={
+                'grant_type': 'urn:ibm:params:oauth:grant-type:apikey',
+                'response_type': 'cloud_iam',
+                'apikey': MOCK_API_KEY
+            }
+        )
+
+        self.assertEqual(access_token, MOCK_ACCESS_TOKEN)
+        self.assertTrue(m_response.raise_for_status.called)
+        self.assertTrue(m_response.json.called)
+
+    @mock.patch('cloudant._common_util.ClientSession.request')
+    @mock.patch('cloudant._common_util.IAMSession._get_access_token')
+    def test_iam_login(self, m_token, m_req):
+        m_token.return_value = MOCK_ACCESS_TOKEN
+        m_response = mock.MagicMock()
+        m_req.return_value = m_response
+
+        iam = IAMSession(MOCK_API_KEY, 'http://127.0.0.1:5984')
+        iam.login()
+
+        m_req.assert_called_once_with(
+            'POST',
+            iam._session_url,
+            headers={'Content-Type': 'application/json'},
+            data=json.dumps({'access_token': MOCK_ACCESS_TOKEN})
+        )
+
+        self.assertEqual(m_token.call_count, 1)
+        self.assertTrue(m_response.raise_for_status.called)
+
+    def test_iam_logout(self):
+        iam = IAMSession(MOCK_API_KEY, 'http://127.0.0.1:5984')
+        # add a valid cookie to jar
+        iam.cookies.set_cookie(self._mock_cookie())
+        self.assertEqual(len(iam.cookies.keys()), 1)
+        iam.logout()
+        self.assertEqual(len(iam.cookies.keys()), 0)
+
+    @mock.patch('cloudant._common_util.ClientSession.get')
+    def test_iam_get_session_info(self, m_get):
+        m_info = {'ok': True, 'info': {'authentication_db': '_users'}}
+
+        m_response = mock.MagicMock()
+        m_response.json.return_value = m_info
+        m_get.return_value = m_response
+
+        iam = IAMSession(MOCK_API_KEY, 'http://127.0.0.1:5984')
+        info = iam.info()
+
+        m_get.assert_called_once_with(iam._session_url)
+
+        self.assertEqual(info, m_info)
+        self.assertTrue(m_response.raise_for_status.called)
+
```

Comment:
I feel like there is a golden path test missing here, checking that a request is as expected without any expiry/renewal conditions.

I also think it would be valuable to have a test at the `client` level verifying that using the class method creates the IAM session and appropriate golden path is followed making a normal request (token, iam_session, normal request).